engine/src/engine.rs

```diff
@@ -201,6 +200,30 @@ impl From<BalanceOverflow> for GasPaymentError {
     }
 }
 
+pub struct ERC20Address(Address);
+impl AsRef<[u8]> for ERC20Address {
+    fn as_ref(&self) -> &[u8] {
+        self.0.as_bytes()
+    }
+}
+impl From<Vec<u8>> for ERC20Address {
+    fn from(bytes: Vec<u8>) -> Self {
+        Self(Address::from_slice(&bytes))
+    }
+}
```

Comment:
Shouldn't this be a TryFrom since the bytes could be more or less? We should try to catch these.

Reply:
Done in https://github.com/aurora-is-near/aurora-engine/pull/342/commits/339dc43def5853776090c5ff17d65f8d788bbd3a